inc/class-registry.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Component registry.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Components;
+
+/**
+ * Get an instance of the registry class.
+ *
+ * @return \WP_Irving\Components\Registry
+ */
+function get_registry() {
+	static $registry;
+
+	if ( empty( $registry ) ) {
+		$registry = new Registry();
+	}
+
+	return $registry;
+}
+
+/**
+ * Auto load some components which will then get registered. Defaults to the
+ * theme directory and WP Irving plugin.
+ */
+function auto_register_components() {
+
+	/**
+	 * Filter the directories from which components should be autoloaded.
+	 *
+	 * @param array $directories Array of directories.
+	 */
+	$directories = apply_filters(
+		'wp_irving_component_registry_directories',
+		[
+			'wp_irving' => WP_IRVING_PATH . '/inc/components/', // Load components from WP Irving.
+			'theme'     => get_stylesheet_directory() . '/components/', // Load components from the activated theme.
+		]
+	);
+
+	get_registry()->load_components( $directories );
+}
+add_filter( 'init', __NAMESPACE__ . '\auto_register_components' );
+
+/**
+ * Registry.
+ */
+class Registry {
+
+	/**
+	 * Array of registered components.
+	 *
+	 * @var array
+	 */
+	protected $components = [];
+
+	/**
+	 * Get all registered components.
+	 *
+	 * @return array
+	 */
+	public function get_registered_components(): array {
+		return $this->components;
+	}
+
+	/**
+	 * Get a single registered component by name.
+	 *
+	 * @param string $name Component name.
+	 * @return array
+	 */
+	public function get_registered_component( string $name ): ?array {
+		return $this->components[ $name ] ?? null;
+	}
+
+	/**
+	 * Register a component.
+	 *
+	 * @param string $name Component name.
+	 * @param array  $args Component args.
+	 */
+	public function register_component( string $name, array $args = [] ) {
+		$this->components[ $name ] = $args;
+	}
+
+	/**
+	 * Register a component using a json config.
+	 *
+	 * @param string $json_file JSON config file.
```

Comment:
Is this a full path?

Reply:
Updated the param to clarify what it is.